tests/fuzzer.c

```diff
@@ -1999,14 +1999,14 @@ static int basicUnitTests(U32 const seed, double compressibility)
 
     /* long rle test */
     {   size_t sampleSize = 0;
+        size_t expectedCompressedSize = 39; /* block 1: RLE, block 2: compressed, block 3: RLE, zstd 1.4.4 */
```

Comment:
Block 1 shouldn't be RLE, are you sure that it is encoded as RLE? It should be block1 compressed, block2 RLE.

Reply:
It shouldn't be RLE because we should be disallowing the first block to be RLE, because of a decoder bug.

Reply:
I added the following check in `ZSTD_compressBlock_internal()` to look for RLE:
```
if (ZSTD_isRLE(ip, srcSize)) fprintf(stderr, "RLE\n");
else fprintf(stderr, "notRLE\n");
```
But just noticed the fix at the bottom to prevent emitting first block as RLE, so yeah, the first block wouldn't be RLE. 

If the second block is indeed just 128KB of 'B', then I suppose the compressor should make it RLE, but according to the `ZSTD_isRLE()` function, which looks like it determines whether or not an input block will be compressed into an RLE block, it returns false on the second block during this test.

Reply:
The second block seems to consists of 128 KB __- 1 bytes__ of 'B' and then one lone 'A', preventing it from being encoded as RLE.

If that's not what you want, you could change the size of 'B' section to be exactly 128 KB instead (256 KB in total with first block).

Reply:
Aha, my mistake was remembering the compression format wrong: I thought that the compressor read the input stream backwards for some reason (in addition to the decompressor), which would mean: 
`block 1: 96KB 'A' + 32KB 'B'`, `block 2: 128KB 'B'`, and `block 3: 96KB - 1byte 'B'`, and therefore block 2-3 are RLE. But read the correct way, then block 3 should be the only RLE block.

As long as one block is RLE, the test will still serve its purpose.